src/main/java/org/colonelkai/publictransit/TravelManager.java

```diff
@@ -0,0 +1,115 @@
+package org.colonelkai.publictransit;
+
+import org.colonelkai.publictransit.line.travel.Travel;
+import org.core.TranslateCore;
+import org.core.entity.living.human.player.User;
+import org.core.schedule.Scheduler;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+public class TravelManager {
+    private final HashMap<Travel, Scheduler> travelMap = new HashMap<>();
+
+    private final Consumer<Scheduler> consumer = (Scheduler s) -> {
+        // check if we're still on the list, or if we've been removed.
+        Optional<Travel> optionalTravel = PublicTransit.getPlugin().getTravelManager().getTravelFromScheduler(s);
+        if (optionalTravel.isEmpty()) {
+            s.cancel(); // double tapping
+        } else {
+            optionalTravel.get().travelToNext();
+        }
+    };
+
+    public Optional<Travel> getTravelFromPlayer(UUID uuid) {
+        return this.travelMap
+                .keySet()
+                .parallelStream()
+                .filter(
+                        travel -> travel.getPlayer().equals(uuid)
+                )
+                .findAny();
+    }
+
+    public Optional<Travel> getTravelFromPlayer(User player) {
+        return this.getTravelFromPlayer(player.getUniqueId());
+    }
+
+    private Optional<Travel> getTravelFromScheduler(Scheduler scheduler) {
+        return this.travelMap.entrySet()
+                .parallelStream()
+                .filter((e) -> {return e.getValue().equals(scheduler);})
```

Comment:
Single line body (either make it a multi-lined body or a single line lamda)

multi-lined body
```java
filter(e -> {
  return e.getValue().equals(scheduler);
})
```

single-lined lamda

```java
filter(e -> e.getValue().equals(scheduler))
```